### PR TITLE
Support versionAsOf and timestampAsOf for getMetadata in spark

### PR DESCRIPTION
**Pull request by linzhou-db**
Support versionAsOf and timestampAsOf for getMetadata in spark, updated existing unit tests.

Please only review changes under spark/ in this PR, all others should be covered by https://github.com/delta-io/delta-sharing/pull/377

### CODE REVIEW

File: spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala [hunk 1]
```diff
@@ -16,24 +16,10 @@
...
-  DeltaTableFiles,
-  DeltaTableMetadata,
-  Metadata,
-  Protocol,
-  RemoveFile,
-  SingleAction,
-  Table
-}
+import io.delta.sharing.client.{DeltaSharingClient, DeltaSharingProfile, DeltaSharingProfileProvider}
+import io.delta.sharing.client.model.{AddCDCFile, AddFile, AddFileForCDF, DeltaTableFiles, DeltaTableMetadata, Metadata, Protocol, RemoveFile, SingleAction, Table}
```
Comment (charlenelyu-db) [inline spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala:20]: nit: format

**Review by charlenelyu-db — APPROVED** [review]

File: spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala (hunk 1 above)
Comment (linzhou-db) [reply spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala]: updated.

File: spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala [hunk 2]
```diff
@@ -56,15 +57,33 @@ class TestDeltaSharingClient(
...
       table: Table,
       versionAsOf: Option[Long] = None,
       timestampAsOf: Option[String] = None): DeltaTableMetadata = {
-    DeltaTableMetadata(0, Protocol(0), metadata)
+    if (versionAsOf.exists(_ == 1)) {
+      DeltaTableMetadata(1, Protocol(0), metadataV1)
+    } else if (timestampAsOf.exists(_ == TESTING_TIMESTAMP)) {
+      DeltaTableMetadata(1, Protocol(0), metadataV2)
+    } else {
+      DeltaTableMetadata(0, Protocol(0), metadata)
```
Comment (chakankardb) [inline spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala:85]: Consider documenting what this function returns.

File: spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala [hunk 3]
```diff
@@ -97,7 +116,13 @@ class TestDeltaSharingClient(
       ).take(limit.getOrElse(4L).toInt)
     }
 
-    DeltaTableFiles(0, Protocol(0), metadata, addFiles)
+    if (versionAsOf.exists(_ == 1)) {
+      DeltaTableFiles(0, Protocol(0), metadataV1, addFiles)
+    } else if (timestampAsOf.exists(_ == TESTING_TIMESTAMP)) {
+      DeltaTableFiles(0, Protocol(0), metadataV2, addFiles)
```
Comment (chakankardb) [inline spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala:122]: Should these be DeltaTableFiles(1, ...) to make it consistent with getMetadata?

Comment (linzhou-db) [reply spark/src/test/scala/io/delta/sharing/spark/TestDeltaSharingClient.scala:122]: @chakankardb  seems that the version is not used.
But you are right, I'll update it.

**Pull request merged**